yt/frontends/stream/tests/test_stream_amrgrids.py

```diff
@@ -69,3 +69,24 @@ def load_grids():
         load_amr_grids(grid_data, dims, bbox=bbox, periodicity=(0, 0, 0),
                        length_unit=1.0, refine_by=2)
     assert_raises(YTIllDefinedAMR, load_grids)
+
+
+def test_load_despite_rounding_errors_during_grid_construction():
+    # from GH issue 2665
+    grid_data = [
+    dict(left_edge =[0.0, 0.0, 0.0],
+         right_edge=[0.045,0.045,0.045],
+         level=0,
+         dimensions=[16,16,16]),
+    dict(left_edge =[0.0225, 0.0225, 0.0225],
+         right_edge=[0.045, 0.045, 0.045],
+         level=1,
+         dimensions=[16,16,16])
+    ]
+
+    for g in grid_data:
+        g["density"] = np.random.random(g["dimensions"]) * 2 ** g["level"]
+
+    bbox = np.array([[-0.18,0.18],[-0.18,0.18],[-0.18,0.18]])
+    dimensions = [128,128,128]
+    load_amr_grids(grid_data, dimensions, bbox)
```

Comment:
is there an "assert not raises" or something we could put here, to check that it is working?

and, could we add a second line, where we load it a second time, but with the left edge of one of the grids offset by some not-quite-negligible value, just to break it, and do an assert raises?

Reply:
> is there an "assert not raises" or something we could put here, to check that it is working?

Well I think this is the very definition of a passing test, and that it doesn't need a dedicated function.
It could be made more clear with comments though, I'll update this with your suggestion :)

Reply:
actually adding this test made it clear that there's a problem in my fix, since the new test fails.